pulser-core/pulser/channels/base_channel.py

```diff
@@ -55,6 +55,8 @@ class Channel(ABC):
             clock cycle.
         min_duration: The shortest duration an instruction can take.
         max_duration: The longest duration an instruction can take.
+        min_amp_area: The minimum area of a pulse's amplitude waveform
+            (when not zero).
```

Comment:
Oh okay, I don't know why I have never noticed that amplitude has to be positive :unamused: I guess having a negative amplitude is the same as having a positive amplitude and a phase $`\pi`$.
I have the feeling that setting a mininum amplitude is closer to reality, because the laser is limited in amplitude and not in area. 
I was thinking of checking that each 4 points of the sample of the amplitude is above a certain value (for Interpolated/CustomWaveforms).
However, I agree it can induce rounding errors.
Additional questions:
- If we have a minimum magnitude/area for the amplitude, should not we have it also for the detuning ?
- I think I have troubles seeing why we are adding this feature now 😅

Reply:
I'll ask around, I'm not sure myself